### PR TITLE
Make insertBefore and insertAfter more flexible

### PR DESCRIPTION
**Pull request by jwmerrill**
`insertBefore` and `insertAfter` now take a whole `DOMFragment` as an argument instead of a single `Node`. Removes many runtime asserts that were added in c85ce0c.

The asserts were useful while doing the translation to ensure we didn't differ in behavior in any cases where `insert{Before,After}` were called with multi-element jQuery collections, but now that that translation is complete, they aren't needed anymore.

Similarly, lifts the requirement in `.replaceWith()` that `this` is a single `Node` collection.

As a follow-on, refines other `.one()` assertions to either `.oneElement()` or `.oneText()`. `.one()` now has no external callers, and has been renamed to `.oneNode()`.

### CODE REVIEW

File: src/domFragment.ts [hunk 1]
```diff
@@ -321,16 +321,16 @@ class DOMFragment {
...
   replaceWith(children: DOMFragment) {
-    const el = this.one();
-    const parent = el.parentNode;
-    pray('parent is defined', parent);
-    parent.replaceChild(children.toDocumentFragment(), el);
+    const rightEnd = this.ends?.[R];
+    const parent = rightEnd?.parentNode;
+    children.detach();
+    if (!rightEnd || !parent) return this;
+    parent.insertBefore(children.toDocumentFragment(), rightEnd.nextSibling);
```
[the remark this answers is not in the document]
Comment (jwmerrill) [reply src/domFragment.ts:340]: Oh, yeah, you're totally right. Line `331` should be `this.detach()`, not `children.detach()`. A tiny bit worrisome that tests didn't spot that.

[the remark this answers is not in the document]
Comment (jwmerrill) [reply src/domFragment.ts:340]: Or I guess it should do both.

[the remark this answers is not in the document]
Comment (jwmerrill) [reply src/domFragment.ts]: Fixed in a rebase in 8398e60.

Turned out to be trickier than I thought because of https://github.com/desmosinc/mathquill/pull/193/commits/8398e606efa0512fec2facca05a9a0ddf530db80#diff-51e268430a28ce2f10a5e466f19d84c16af8508a00f00ec278d4ec5c1319e6aaR338-R340

**Review by anandthakker — APPROVED** [review]
